plugins/deprecation.nix

```
@@ -64,22 +72,29 @@ in
           Please use `plugins.web-devicons` or `plugins.mini.modules.icons` with `plugins.mini.mockDevIcons` instead.
         ''
     ) iconsPackagePlugins
-    # Show a warning when web-devicons is auto-enabled
     ++ [
       (
         { config, options, ... }:
         {
-          config = lib.mkIf (options.plugins.web-devicons.enable.highestPrio == 1490) {
-            warnings = [
-              ''
-                Nixvim: `plugins.web-devicons` was enabled automatically because the following plugins are enabled.
-                This behaviour is deprecated. Please explicitly define `plugins.web-devicons.enable` or alternatively
-                enable `plugins.mini.enable` with `plugins.mini.modules.icons` and `plugins.mini.mockDevIcons`.
-                ${lib.concatMapStringsSep "\n" (name: "plugins.${name}") (
-                  builtins.filter (name: config.plugins.${name}.enable) iconsPackagePlugins
-                )}
-              ''
-            ];
+          config = {
+            warnings =
+              lib.optionals (options.plugins.web-devicons.enable.highestPrio == 1490) [
+                ''
+                  Nixvim: `plugins.web-devicons` was enabled automatically because the following plugins are enabled.
+                  This behaviour is deprecated. Please explicitly define `plugins.web-devicons.enable` or alternatively
+                  enable `plugins.mini.enable` with `plugins.mini.modules.icons` and `plugins.mini.mockDevIcons`.
+                  ${lib.concatMapStringsSep "\n" (name: "plugins.${name}") (
+                    builtins.filter (name: config.plugins.${name}.enable) iconsPackagePlugins
+                  )}
+                ''
+              ]
+              ++ lib.concatMap (
+                plugin:
+                lib.optional config.plugins."${plugin}".enable ''
+                  Nixvim Warning: The `${plugin}` plugin has been deprecated.
+                  ${deprecated.${plugin}}
+                ''
+              ) (lib.attrNames deprecated);
```

Take it or leave it, no pressure:

```suggestion
              ++ lib.foldAttrs (
                warnings: plugin: msg:
                warnings ++ lib.optional config.plugins.${plugin}.enable ''
                  Nixvim Warning: The `${plugin}` plugin has been deprecated.
                  ${msg}
                ''
              ) [ ] deprecated;
```

I just wonder if it's a bit more readable becaues you don't need the `${deprecated.${plugin}}` or the `(lib.attrNames deprecated)` :thinking:

---

I tried this but kept getting  errors. 
    >  error: expected a list but found a set: { rust-tools = "The `rust-tools` project has been abandoned.\n\nIt is recommended to use `rustaceanvim` instead.\n"; }
Test failure

---

Hm, I just typed it out on github, so may have errors. I'll check real quick

---

My bad, I forgot that [`foldAttrs`](https://noogle.dev/f/lib/foldAttrs) is different to [`foldlAttrs`](https://noogle.dev/f/lib/foldlAttrs):

```suggestion
              ++ lib.foldlAttrs (
                warnings: plugin: msg:
                warnings ++ lib.optional config.plugins.${plugin}.enable ''
                  Nixvim Warning: The `${plugin}` plugin has been deprecated.
                  ${msg}
                ''
              ) [ ] deprecated;
```

```
nix-repl> :lf .
Added 25 variables.

nix-repl> lib = legacyPackages.x86_64-linux.nixvimConfiguration._module.specialArgs.lib

nix-repl> config = legacyPackages.x86_64-linux.nixvimConfiguration.config

nix-repl> lib.foldAttrs (warnings: name: msg: warnings ++ lib.optional (true) "Test ${name} - ${msg}") [ ] { foo = "bar"; bar = "baz"; }
error:
       … while evaluating a branch condition
         at /nix/store/w7nfn4lcfisddz14b0y8n05rjf0dl5lq-source/lib/lists.nix:125:9:
          124|       fold' = n:
          125|         if n == len
             |         ^
          126|         then nul

       … while calling the 'length' builtin
         at /nix/store/w7nfn4lcfisddz14b0y8n05rjf0dl5lq-source/lib/lists.nix:123:13:
          122|     let
          123|       len = length list;
             |             ^
          124|       fold' = n:

       (stack trace truncated; use '--show-trace' to show the full, detailed trace)

       error: expected a list but found a set: { bar = "baz"; foo = "bar"; }

nix-repl> lib.foldlAttrs (warnings: name: msg: warnings ++ lib.optional (true) "Test ${name} - ${msg}") [ ] { foo = "bar"; bar = "baz"; }
[
  "Test bar - baz"
  "Test foo - bar"
]

```

---

Cool, that works. Thanks!